skbio/diversity/alpha/_faith_pd.py

```diff
@@ -0,0 +1,158 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from skbio.util._decorator import experimental
+from skbio.diversity._util import (_validate_counts_vector,
+                                   _validate_otu_ids_and_tree,
+                                   _vectorize_counts_and_tree)
+
+
+def _faith_pd(counts_by_node, branch_lengths):
+    observed_nodes = np.where(counts_by_node > 0, 1, 0)
+    return (branch_lengths * observed_nodes).sum()
```

Comment:
fixed, we had a variation on that using ``nonzero`` that wasn't working right, and then I kind of forgot about it. it's in there now (locally).